test/react-native/features/session.feature

```diff
@@ -25,142 +25,143 @@ Scenario: Automatic sessions disabled
-  And the payload field "app" is not null
-  And the payload field "device" is not null
-  And the payload has a valid sessions array
-  And the payload field "sessions.0.id" is stored as the value "initial_session_id"
-
-  And I discard the oldest request
+  Then I wait to receive 2 sessions
+  And I wait to receive 4 errors
+
+  Then the session "bugsnag-api-key" header equals "12312312312312312312312312312312"
+  And the session "bugsnag-payload-version" header equals "1.0"
+  And the session "Content-Type" header equals "application/json"
+  And the session "Bugsnag-Sent-At" header is a timestamp
+  And the session payload field "notifier.name" equals "Bugsnag React Native"
+  And the session payload field "notifier.url" equals "https://github.com/bugsnag/bugsnag-js"
+  And the session payload field "notifier.version" is not null
+  And the session payload field "app" is not null
+  And the session payload field "device" is not null
+  And the session payload has a valid sessions array
+  And the session payload field "sessions.0.id" is stored as the value "initial_session_id"
+
+  And I discard the oldest session
 
   And the exception "errorClass" equals "Error"
   And the exception "message" equals "SessionJsControlledManualJsScenarioA"
   And the event "unhandled" is false
   And the event "session" is not null
-  And the payload field "events.0.session.id" equals the stored value "initial_session_id"
+  And the error payload field "events.0.session.id" equals the stored value "initial_session_id"
   And the event "session.events.handled" equals 1
 
-  And I discard the oldest request
+  And I discard the oldest error
 
   And the exception "errorClass" equals "Error"
   And the exception "message" equals "SessionJsControlledManualJsScenarioB"
   And the event "unhandled" is false
   And the event "session" is null
 
-  And I discard the oldest request
+  And I discard the oldest error
 
   And the exception "errorClass" equals "Error"
   And the exception "message" equals "SessionJsControlledManualJsScenarioC"
   And the event "unhandled" is false
   And the event "session" is not null
-  And the payload field "events.0.session.id" equals the stored value "initial_session_id"
+  And the error payload field "events.0.session.id" equals the stored value "initial_session_id"
   And the event "session.events.handled" equals 2
 
-  And I discard the oldest request
+  And I discard the oldest error
 
-  And the payload has a valid sessions array
-  And the payload field "sessions.0.id" does not equal the stored value "initial_session_id"
-  And the payload field "sessions.0.id" is stored as the value "second_session_id"
-
-  And I discard the oldest request
+  And the session payload has a valid sessions array
+  And the session payload field "sessions.0.id" does not equal the stored value "initial_session_id"
+  And the session payload field "sessions.0.id" is stored as the value "second_session_id"
```

Comment:
I think it would be cleaner to move these expectations on the second session to immediately follow the first. Then we're checking all sessions, then checking all errors - without saying anything about the order that we might expect between them.

Reply:
I did that initially, but thought it would be more readable to follow the order in which sessions/errors were delivered:
- First session received
- Error matches first session
- Error matches no session (session is paused)
- Error matches first session (session is resumed)
- Second session received
- Error matches second session

Reply:
Yeah, that's fair - I take the point that there is a logical ordering there and it's not about the order in which requests are received as such.

Reply:
I've added some comments to make this ordering clearer